solidity/contracts/hooks/libs/AbstractMessageIdAuthHook.sol

```diff
@@ -82,11 +86,19 @@ abstract contract AbstractMessageIdAuthHook is
             metadata.msgValue(0) < 2 ** 255,
             "AbstractMessageIdAuthHook: msgValue must be less than 2 ** 255"
         );
+
         bytes memory payload = abi.encodeCall(
             AbstractMessageIdAuthorizedIsm.verifyMessageId,
             id
         );
         _sendMessageId(metadata, payload);
+
+        uint256 _overpayment = msg.value - _quoteDispatch(metadata, message);
+        if (_overpayment > 0) {
+            address _refundAddress = metadata.refundAddress(msg.sender);
+            require(_refundAddress != address(0), "no refund address");
+            payable(_refundAddress).sendValue(_overpayment);
+        }
```

Comment:
hm, I dont like the fact that `_sendMessageId` could pass all of the `msg.value` to the bridge and then this would revert
I also dont think defaulting to the `msg.sender` for the refund address is a good idea, wouldnt that often be the mailbox/another hook?

Reply:
A. I think the _sendMessageId should pass quoteDispatch not msg.value

B. What's an alternative here?

Reply:
the hyperlane `message.sender()` like we do elsewhere

Reply:
fixed here: https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/4552/commits/e16958ea5e754c1541eb508e1beb53fdc49224b3